File: OPENAPI-REST-API/openapi-client/clojure/src/inter_server_management_api/specs/domain_service_info.clj
(ns inter-server-management-api.specs.domain-service-info
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def domain-service-info-data
  {
   (ds/opt :domain_id) string?
   (ds/opt :domain_hostname) string?
   (ds/opt :domain_username) string?
   (ds/opt :domain_password) string?
   (ds/opt :domain_type) string?
   (ds/opt :domain_currency) string?
   (ds/opt :domain_expire_date) string?
   (ds/opt :domain_order_date) string?
   (ds/opt :domain_custid) string?
   (ds/opt :domain_status) string?
   (ds/opt :domain_invoice) string?
   (ds/opt :domain_coupon) string?
   })

(def domain-service-info-spec
  (ds/spec
    {:name ::domain-service-info
     :spec domain-service-info-data}))

File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/DomainServiceInfo.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import DomainServiceInfo._

case class DomainServiceInfo (
  domain_id: Option[String],
domain_hostname: Option[String],
domain_username: Option[String],
domain_password: Option[String],
domain_type: Option[String],
domain_currency: Option[String],
domain_expire_date: Option[String],
domain_order_date: Option[String],
domain_custid: Option[String],
domain_status: Option[String],
domain_invoice: Option[String],
domain_coupon: Option[String])

object DomainServiceInfo {
  import DateTimeCodecs._

  implicit val DomainServiceInfoCodecJson: CodecJson[DomainServiceInfo] = CodecJson.derive[DomainServiceInfo]
  implicit val DomainServiceInfoDecoder: EntityDecoder[DomainServiceInfo] = jsonOf[DomainServiceInfo]
  implicit val DomainServiceInfoEncoder: EntityEncoder[DomainServiceInfo] = jsonEncoderOf[DomainServiceInfo]
}
